### PR TITLE
Allow assigning CSS variables in safecss_filter_attr

### PR DESCRIPTION
**Pull request by aristath**
Trac ticket: https://core.trac.wordpress.org/ticket/56353

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/kses.php [hunk 1]
```diff
@@ -2426,6 +2428,10 @@ function safecss_filter_attr( $css, $deprecated = '' ) {
 			$parts        = explode( ':', $css_item, 2 );
 			$css_selector = trim( $parts[0] );
 
+			if ( in_array( '--*', $allowed_attr, true ) && 0 === strpos( $css_selector, '--' ) && sanitize_key( $css_selector ) === $css_selector ) {
```
Comment (ramonjd) [inline src/wp-includes/kses.php]: Would it be an issue that `sanitize_key` forces lower case, so `--HOORAH` becomes `--hoorah`?

Also `in_array( '--*', $allowed_attr, true )` is always going to be `true`, no? 

What if we checked for `--` at the start of the `$css_selector`? I was playing with [something like this over in this PR](https://github.com/WordPress/wordpress-develop/pull/3080/files#diff-13e853d5729978a0e7936543d65ad1a3e120b994702c9559bf99c3ad857f5066R2431)

`if ( strpos( $css_selector, '--' ) === 0 && preg_match( '/^([a-z0-9\-]+)$/', $css_selector ) ) {`

Comment (aristath) [reply src/wp-includes/kses.php]: > Would it be an issue that `sanitize_key` forces lower case, so `--HOORAH` becomes `--hoorah`?

Thank you for the suggestion! Pushed a fix based on the code you posted 👍 

> Also `in_array( '--*', $allowed_attr, true )` is always going to be `true`, no?

Not necessarily... `$allowed_attr` is filterable, so someone may choose to disable css-vars by removing it from the array.

File: src/wp-includes/kses.php [hunk 2]
```diff
@@ -2419,18 +2421,29 @@ function safecss_filter_attr( $css, $deprecated = '' ) {
...
+			}
+
 			if ( in_array( $css_selector, $allowed_attr, true ) ) {
 				$found         = true;
 				$url_attr      = in_array( $css_selector, $css_url_data_types, true );
 				$gradient_attr = in_array( $css_selector, $css_gradient_data_types, true );
 			}
+
+			if ( $is_custom_var ) {
+				$url_attr      = 0 === strpos( $parts[1], 'url(' ) );
```
Comment (SergeyBiryukov) [inline src/wp-includes/kses.php]: I think there's a PHP syntax error here, one opening bracket and two closing.

Comment (aristath) [reply src/wp-includes/kses.php]: thank you! fixed 👍

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in [r54117](https://core.trac.wordpress.org/changeset/54117).

**Pull request closed without merging**